src/vm-repair/azext_vm_repair/custom.py

```diff
@@ -650,3 +665,64 @@ def reset_nic(cmd, vm_name, resource_group_name, yes=False):
         return_dict = command.init_return_dict()
 
     return return_dict
+
+
+def repair_and_restore(cmd, vm_name, resource_group_name, repair_password=None, repair_username=None, repair_vm_name=None, copy_disk_name=None, repair_group_name=None):
+    from datetime import datetime
+    import secrets
+    import string
+
+    password_length = 30
+    password_characters = string.ascii_lowercase + string.digits + string.ascii_uppercase
+    repair_password = ''.join(secrets.choice(password_characters) for i in range(password_length))
+
+    username_length = 20
+    username_characters = string.ascii_lowercase + string.digits
+    repair_username = ''.join(secrets.choice(username_characters) for i in range(username_length))
+
+    timestamp = datetime.utcnow().strftime('%Y%m%d%H%M%S')
+    repair_vm_name = ('repair-' + vm_name)[:14] + '_'
+    copy_disk_name = vm_name + '-DiskCopy-' + timestamp
+    repair_group_name = 'repair-' + vm_name + '-' + timestamp
+    existing_rg = _check_existing_rg(repair_group_name)
+
+    create_out = create(cmd, vm_name, resource_group_name, repair_password, repair_username, repair_vm_name=repair_vm_name, copy_disk_name=copy_disk_name, repair_group_name=repair_group_name, associate_public_ip=False, yes=True)
+
+    # log create_out
+    logger.info('create_out: %s', create_out)
+
+    repair_vm_name = create_out['repair_vm_name']
+    copy_disk_name = create_out['copied_disk_name']
+    repair_group_name = create_out['repair_resource_group']
+
+    logger.info('Running fstab run command')
+
+    try:
+        run_out = run(cmd, repair_vm_name, repair_group_name, run_id='linux-alar2', parameters=["fstab"])
+
+    except Exception:
+        logger.info('fstab run command errored')
```

Comment:
May I ask if the log should be `error` or `warning` instead of `info`?
Besides,  it is suggested to specify the reason of failure.

Reply:
Added command.error_message